src/unix/fs.c

```diff
@@ -858,9 +858,14 @@ static ssize_t uv__fs_buf_iter(uv_fs_t* req, uv__fs_buf_iter_processor process)
     total += result;
   }
 
+  if (errno == EINTR && total == -1)
+    return total;
```

Comment:
I am really curious about how it will work if the loop will terminate after one iteration.